rust/cubesql/cubesql/src/compile/snapshots/cubesql__compile__tests__test_thought_spot_yearly_plan.snap

```diff
@@ -0,0 +1,20 @@
+---
+source: cubesql/src/compile/mod.rs
+assertion_line: 6442
+expression: "format!(\"{:?}\", logical_plan)"
+---
+Projection: #CAST(CAST(datepart(Utf8("YEAR"),ta_1.order_date) * Int64(100) + Int64(1) * Int64(100) + Int64(1) AS Utf8) AS Date32) AS ca_1, CASE WHEN #SUM(ta_1.count) IS NOT NULL THEN #SUM(ta_1.count) ELSE Int64(0) END AS ca_2
```

Comment:
This assertation is important, because without this fix SQL API can reassembly this query by splitting/pushing all expression for outer_aggregate, but it's doesn't work.

>Execution error: Arrow error: Cast error: Cannot cast string '20190101' to value of arrow::datatypes::types::Date32Type type

Reply:
Looks like Postgresql supports different delimiters for casting:

```
select cast('2019|01|01' as date);
select cast('2019\01\01' as date);
select cast('2019/01/01' as date);
```